x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/pipelines/ml_inference/add_ml_inference_button.tsx

```diff
@@ -25,8 +26,10 @@ export const AddMLInferencePipelineButton: React.FC<AddMLInferencePipelineButton
   const { capabilities } = useValues(KibanaLogic);
   const { ingestionMethod } = useValues(IndexViewLogic);
   const { canUseMlInferencePipeline, hasIndexIngestionPipeline } = useValues(PipelinesLogic);
+  const { hasPlatinumLicense } = useValues(LicensingLogic);
   const hasMLPermissions = capabilities?.ml?.canAccessML ?? false;
-  if (!hasMLPermissions) {
+
+  if (!hasMLPermissions || !hasPlatinumLicense) {
```

Comment:
Will this still work for trials?

Reply:
In my testing trial licenses don't fall under platinum.

Reply:
You need a platinum license to add a model to the cluster, which is why I think this is technically redundant. Not sure if you can add a model with a trial license, I will test that and see how it interacts with this check @kderusso

Reply:
@sphilipse @kderusso  `hasPlatinumLicense === true` with a trial license (at least locally)